services/web/server/src/simcore_service_webserver/projects/projects_api.py

```diff
@@ -297,15 +296,10 @@ async def update_project_node_state(
         project_id,
         user_id,
     )
-    project = await get_project_for_user(app, project_id, user_id)
-    if not node_id in project["workbench"]:
-        raise NodeNotFoundError(project_id, node_id)
-    if project["workbench"][node_id].get("state", {}).get("currentStatus") == new_state:
-        # nothing to do here
-        return project
-    project["workbench"][node_id].setdefault("state", {}).update(
-        {"currentStatus": new_state}
-    )
+    project = {
```

Comment:
So instead of getting the project from the db, you create a "thin" project from scratch with the data you have in place.
A small inconvenient is that this "thin" project might get outdated with respect to the project schema. 
perhaps an ``assert `` validator might help? It iwll be remved in production

Reply:
hmm I am not sure I agree with your statement.
What I do is that I pass only the things I want to modify in the project (thus patching). I will create a separate function for this maybe. I have been tricked by the fact that the fonction was originally called update_project... altough it does not that but completely replaces the project.

and I will add some tests... no worries... the current ones are failing for that very reason.

Reply:
tests added. so to answer your question:
- you patch a project workbench, so you only bring what needs to be changed and that is all. no thin project thingy.